### PR TITLE
Add GitHub actions

### LINKED ISSUE #2: Run tests in github actions
Use github actions to run tests (tslint, integration tests).

Documentation can be found here: https://docs.github.com/en/actions

Note: we are using github actions at CodeChecker: https://github.com/Ericsson/codechecker/blob/master/.github/workflows/test.yml. This can be an example how we can use it in this repository too.

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #2

### CODE REVIEW

**Comment by csordasmarton** [comment]
@Discookie I added the linter tests to the github actions so it will run tests on every pull requests. I also enabled some more rules (e.g.: to prevent long lines, use consistent style code, etc.). Please review it.

**Review by Discookie — CHANGES_REQUESTED** [review]
A larger correction, and a couple of suggestions.

Overall I'm not sure if using ESLint for stylistic linting is a good idea, a tool like [Prettier] is much better suited for the job.
Maybe expanding the linter's checks should be its own PR/issue, separate from this?

[Prettier]: https://prettier.io/

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  lint:
+    name: Linters
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - name: Install dependencies
+        run: npm install
```
[the remark this answers is not in the document]
Comment (csordasmarton) [reply .github/workflows/test.yml]: Okay, good point, I changed it to yarn.

File: .eslintrc.json [hunk 2]
```diff
@@ -13,8 +13,16 @@
         "@typescript-eslint/semi": "warn",
         "curly": "warn",
         "eqeqeq": "warn",
+        "indent": ["error", 4],
+        "keyword-spacing": ["error"],
+        "max-len": ["error", { "code": 80 }],
```
[the remark this answers is not in the document]
Comment (csordasmarton) [reply .eslintrc.json]: I checked it and most of the users use 120 max length for TypeScript. So I changed it to 120 too.

By the way if it is not enough, we can save some extra space if we don't use 4 spaces for indentation but 2.

File: .eslintrc.json [hunk 3]
```diff
@@ -13,8 +13,16 @@
         "@typescript-eslint/semi": "warn",
         "curly": "warn",
         "eqeqeq": "warn",
+        "indent": ["error", 4],
```
[the remark this answers is not in the document]
Comment (csordasmarton) [reply .eslintrc.json:16]: I changed all the reports to errors.

**Comment by csordasmarton** [comment]
For now I think one tool is enough for us and I recommend to use ESLint. Later we can introduce Prettier if we still want.

**Review by Discookie — APPROVED** [review]
Sure thing, LGTM!

**Pull request merged**